### PR TITLE
Remove non-inline modules in blocks

### PR DESCRIPTION
**Pull request by jseyfried**
r? @alexcrichton

### CODE REVIEW

File: src/bin/cargo.rs [hunk 1]
```diff
@@ -91,7 +91,14 @@ macro_rules! each_subcommand{
...
         $mac!(yank);
-    })
+    }
+}
+
+mod subcommands {
+    macro_rules! declare_mod {
+        ($name:ident) => ( pub mod $name; )
+    }
+    each_subcommand!(declare_mod);
```
Comment (alexcrichton) [inline src/bin/cargo.rs]: Weird, I'm kinda surprised that this doesn't require the path to the source file to be `src/bin/subcommands/foo.rs`...

**Comment by alexcrichton** [comment]
@bors: r+ d521a9a007c596a8fdbdfbac5a812f789c145ac5

Also for onlookers, this is to ward of the upcoming breakage of https://github.com/rust-lang/rust/pull/31534

File: src/bin/cargo.rs (hunk 1 above)
Comment (jseyfried) [inline src/bin/cargo.rs]: Yeah, it should require that. Right now, macro-expanded non-inline module declarations are always treated as if they were at the file root due to a bug in the macro expander.

Comment (alexcrichton) [inline src/bin/cargo.rs]: wheeeeeee

**Pull request merged**